src/Meditation.MetadataLoaderService/Models/MethodMetadataEntry.cs

```diff
@@ -1,7 +1,29 @@
-﻿using System.Collections.Immutable;
+﻿using dnlib.DotNet;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
 
 namespace Meditation.MetadataLoaderService.Models
 {
-    public record MethodMetadataEntry(string Name, MethodDefinitionToken MethodDefinitionToken, ImmutableArray<MetadataEntryBase> Children)
-        : MetadataEntryBase(Name, new MetadataToken(MethodDefinitionToken.Value), Children);
+    public record MethodMetadataEntry : MetadataEntryBase
+    {
+        internal readonly MethodDef MethodDef;
+
+        internal MethodMetadataEntry(MethodDef methodDef, ImmutableArray<MetadataEntryBase> children)
+            : base(methodDef.Name, new MetadataToken(methodDef.MDToken.ToInt32()), children)
+        {
+            MethodDef = methodDef;
+        }
+
+        public string DeclaringTypeFullName => MethodDef.DeclaringType.ReflectionFullName;
+        public IEnumerable<string> ParameterTypeFullNames => MethodDef.Parameters.Where(p => p.IsNormalMethodParameter).Select(p => p.Type.ReflectionFullName);
```

Comment:
[ToFix] Start every method with lazy enumeration with `Enumerate`. Avoid using properties with `IEnumerable` type because you don't want to get IO/lock/heavy calculation problems during their calls. For example during serialization, logging, etc.

Make a method `EnumerateParameterTypeFullNames()` if you need lazy enumeration. So anyone will see that it's bad idea to call it cases like above.